crates/torii/indexer/src/engine.rs

```diff
@@ -549,8 +549,13 @@ impl<P: Provider + Send + Sync + std::fmt::Debug + 'static> Engine<P> {
                     let last_block_number =
                         events_page.events.last().map_or(0, |e| e.block_number.unwrap());
 
-                    // Process events for this page
+                    // Process events for this page, only including events up to our target block
                     for event in events_page.events {
+                        let block_number = event.block_number.unwrap();
+                        if block_number > to {
```

Comment:
Isn't the block number already used in the filter when events are fetched? Why should we test again here?